File: src/Java/401-500/401.BinaryWatch.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.Queue;
import java.util.Set;
import java.util.HashSet;
import java.util.HashMap;
import java.util.LinkedList;

class Solution {
    public int bit1count(String s){
        int count = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '1')
                ++count;
        }
        return count;
    }

    public List<String> readBinaryWatch(int num) {
        LinkedList<String> times = new LinkedList<>();
        for (int i = 0; i < 12; i++) {
            for (int j = 0; j < 60; j++) {
                int h_count = bit1count(Integer.toBinaryString(i));
                int m_count = bit1count(Integer.toBinaryString(j));
                if (h_count + m_count == num)
                    times.add(String.format("%d:%02d", i, j));
            }
        }
        return times;
    }
}

File: src/Python/401-500/401.BinaryWatch.py

class Solution:
    def readBinaryWatch(self, num):
        """
        :type num: int
        :rtype: List[str]
        """
        times = []
        for i in range(12):
            for j in range(60):
                if bin(i).count('1') + bin(j).count('1') == num:         
                    times.append('{:d}:{:02d}'.format(i, j))
        return times

if __name__ == '__main__':
    solution = Solution()
    print(solution.readBinaryWatch(0))
    print(solution.readBinaryWatch(1))
    print(solution.readBinaryWatch(2))
    print(solution.readBinaryWatch(3))
    print(solution.readBinaryWatch(4))
    print(solution.readBinaryWatch(5))
    print(solution.readBinaryWatch(8))
    print(solution.readBinaryWatch(9))
    print(solution.readBinaryWatch(10))
else:
    pass
